protocols/v2/codec-sv2/src/encoder.rs

```diff
@@ -1,3 +1,26 @@
+//! # Encoder
+//!
+//! Provides utilities for encoding messages into Sv2 frames, with or without Noise protocol
+//! support.
+//!
+//! ## Usage
+//!
+//! All messages passed between Sv2 roles are encoded as Sv2 frames using primitives in this
+//! module. There are two types of encoders for creating these frames: one for regular Sv2 frames
+//! (`Encoder`), and another for Noise-encrypted frames (`NoiseEncoder`). Both encoders manage the
+//! serialization of outgoing data and, when applicable, the encryption of the data before
+//! transmission.
+//!
+//! ### Buffer Management
+//!
+//! The encoders rely on buffers to hold intermediate data during the encoding process.
+//!
+//! - When the `with_buffer_pool` feature is enabled, the internal `Buffer` type is backed by a
+//!   pool-allocated buffer (`binary_sv2::BufferPool`), providing more efficient memory usage,
+//!   particularly in high-throughput scenarios.
+//! - If the feature is not enabled, a system memory buffer (`binary_sv2::BufferFromSystemMemory`)
+//!   is used for simpler applications where memory efficiency is less critical.
```

Comment:
@fi3, can you review my buffer understanding?

Reply:
`BufferPool` uses unsafe Rust, which is probably an important detail to be mentioned here.

Reply:
aside from this detail on unsafe Rust, the description we have here is pretty aligned with my current understanding of `with_buffer_pool`

Reply:
another improvement here is that we could embed these links:
- https://docs.rs/buffer_sv2/latest/buffer_sv2/struct.BufferPool.html
- https://docs.rs/buffer_sv2/latest/buffer_sv2/struct.BufferFromSystemMemory.html

not only here, but also in the other references to `Buffer` and `BufferPool`